exporter/awscloudwatchlogsexporter/config_test.go

```diff
@@ -20,84 +20,87 @@ import (
-
-	assert.Equal(t,
-		&Config{
-			ExporterSettings: config.NewExporterSettings(config.NewComponentIDWithName(typeStr, "e2-no-retries-short-queue")),
-			RetrySettings: exporterhelper.RetrySettings{
-				Enabled:         false,
-				InitialInterval: defaultRetrySettings.InitialInterval,
-				MaxInterval:     defaultRetrySettings.MaxInterval,
-				MaxElapsedTime:  defaultRetrySettings.MaxElapsedTime,
-			},
-			AWSSessionSettings: awsutil.CreateDefaultSessionConfig(),
-			LogGroupName:       "test-2",
-			LogStreamName:      "testing",
-			QueueSettings: QueueSettings{
-				QueueSize: 2,
+		{
+			id: config.NewComponentIDWithName(typeStr, "e2-no-retries-short-queue"),
+			expected: &Config{
+				ExporterSettings: config.NewExporterSettings(config.NewComponentID(typeStr)),
+				RetrySettings: exporterhelper.RetrySettings{
+					Enabled:         false,
+					InitialInterval: defaultRetrySettings.InitialInterval,
+					MaxInterval:     defaultRetrySettings.MaxInterval,
+					MaxElapsedTime:  defaultRetrySettings.MaxElapsedTime,
+				},
+				AWSSessionSettings: awsutil.CreateDefaultSessionConfig(),
+				LogGroupName:       "test-2",
+				LogStreamName:      "testing",
+				QueueSettings: QueueSettings{
+					QueueSize: 2,
+				},
 			},
 		},
-		e2,
-	)
-}
-
-func TestFailedLoadConfig(t *testing.T) {
-	factories, err := componenttest.NopFactories()
-	assert.NoError(t, err)
-
-	factory := NewFactory()
-	factories.Exporters[typeStr] = factory
-
-	_, err = servicetest.LoadConfigAndValidate(filepath.Join("testdata", "missing_required_field_1_config.yaml"), factories)
-	assert.EqualError(t, err, "exporter \"awscloudwatchlogs\" has invalid configuration: 'log_stream_name' must be set")
-
-	_, err = servicetest.LoadConfigAndValidate(filepath.Join("testdata", "missing_required_field_2_config.yaml"), factories)
-	assert.EqualError(t, err, "exporter \"awscloudwatchlogs\" has invalid configuration: 'log_group_name' must be set")
-
-	_, err = servicetest.LoadConfigAndValidate(filepath.Join("testdata", "invalid_queue_size.yaml"), factories)
-	assert.EqualError(t, err, "exporter \"awscloudwatchlogs\" has invalid configuration: 'sending_queue.queue_size' must be 1 or greater")
-
-	_, err = servicetest.LoadConfigAndValidate(filepath.Join("testdata", "invalid_queue_setting.yaml"), factories)
-	assert.EqualError(t, err, "error reading exporters configuration for \"awscloudwatchlogs\": 1 error(s) decoding:\n\n* 'sending_queue' has invalid keys: enabled, num_consumers")
+		{
+			id:           config.NewComponentIDWithName(typeStr, "invalid_queue_size"),
+			errorMessage: "'sending_queue.queue_size' must be 1 or greater",
+		},
+		{
+			id:           config.NewComponentIDWithName(typeStr, "invalid_required_field_stream"),
+			errorMessage: "'log_stream_name' must be set",
+		},
+		{
+			id:           config.NewComponentIDWithName(typeStr, "invalid_required_field_group"),
+			errorMessage: "'log_group_name' must be set",
```

Comment:
it looks like a test for `invalid_queue_setting` is missing here

Reply:
Added it back